cantools/database/can/formats/dbc.py

```diff
@@ -1705,8 +1889,11 @@ def dump_string(database: InternalDatabase, sort_signals: type_sort_signals = SO
                           cm='\r\n'.join(cm),
                           signal_types='\r\n'.join(signal_types),
                           ba_def='\r\n'.join(ba_def),
+                          ba_def_rel="".join([elem+"\r\n" for elem in ba_def_rel]),
```

Comment:
Correct that line endings are handled differently here? I would prefer keeping it consistent if possible.

Reply:
yes, \r\n  should only be added if at least one element present in ba_def_rel. otherwise extra empty spaces will exists in output dbc even though there is no empty space in input. I thought about changing old implementation but I stick to feature that I thought for now. 

It is not possible to implement line endings as done previously.